ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -127,14 +127,15 @@ runLanguageServer recorder options inH outH defaultConfig parseConfig onConfigCh
 setupLSP ::
      forall config err.
      Recorder (WithPriority Log)
+  -> FilePath -- ^ root directory
```

Comment:
For example, this needs to reference a note, explaining what's going on, why we have it, etc...

Reply:
done